sky/clouds/gcp.py

```diff
@@ -131,17 +131,14 @@ def regions(cls) -> List[clouds.Region]:
         return cls._regions
 
     @classmethod
-    def region_zones_provision_loop(
-        cls,
-        *,
-        instance_type: Optional[str] = None,
-        accelerators: Optional[Dict[str, int]] = None,
-        use_spot: Optional[bool] = False,
-    ) -> Iterator[Tuple[clouds.Region, List[clouds.Zone]]]:
-        # GCP provisioner currently takes 1 zone per request.
+    def regions_with_offering(cls, instance_type: Optional[str],
+                              accelerators: Optional[Dict[str, int]],
+                              use_spot: bool, region: Optional[str],
+                              zone: Optional[str]) -> List[clouds.Region]:
         if accelerators is None:
             if instance_type is None:
-                # fallback to manually specified region/zones
+                # Fall back to the default regions.
+                # TODO: Get the regions from the service catalog.
                 regions = cls.regions()
             else:
                 regions = service_catalog.get_region_zones_for_instance_type(
```

Comment:
nit: We have the fallback logic inside the `regions_with_offering` in aws/azure.py. Do we want to have it in the `region_with_offering` for gcp.py as well?

Reply:
What do you mean by the fallback logic in AWS and Azure? I think AWS and Azure also fall back to their default regions.
```python
if instance_type is None:
    # Fall back to default regions
    regions = cls.regions()
```

Reply:
Oops, my bad. It seems I misread the function name. It is good.